webrender/src/renderer.rs

```diff
@@ -414,6 +415,10 @@ pub struct Renderer {
 
     /// Map of external image IDs to native textures.
     external_images: HashMap<ExternalImageId, TextureId, BuildHasherDefault<FnvHasher>>,
+
+    // Optional trait object that allows to create a VRCompositor instance to handle WebVR commands.
+    // Some WebVR commands such as Vsync and SubmitFrame must be called in the WebGL render thread.
+    vr_compositor_creator: Arc<Mutex<Option<Box<VRCompositorCreator>>>>
```

Comment:
This compositor creator is referenced from both the render backend and the renderer. And it looks like the only reason it's needed on the renderer is for `set_vr_compositor_creator`. Can we assume the compositor creator is known at the init time? In this case, we might be able to just send it over and store without all the `Arc`/`Mutex` wrappers.